EIPS/eip-1363.md

```diff
@@ -181,24 +181,27 @@ interface ERC1363Spender {
    * approval. Return of other than the magic value MUST result in the
    * transaction being reverted.
    * Note: the token contract address is always the message sender.
-   * @param owner address The address which called `approveAndCall` function
-   * @param value uint256 The amount of tokens to be spent
+   * @param sender address The address which called `approveAndCall` function
+   * @param amount uint256 The amount of tokens to be spent
    * @param data bytes Additional data with no specified format
    * @return `bytes4(keccak256("onApprovalReceived(address,uint256,bytes)"))`
```

Comment:
I think you should give the easy way to calculate this, using a function selector:

`* @return ERC1363Spender.onApprovalReceived.selector == bytes4(keccak256("onApprovalReceived(address,uint256,bytes)"))`;

Same for the receiver interface too.

Reply:
Uhm yes, right. But `bytes4(keccak256("onApprovalReceived(address,uint256,bytes)"))` will always remain the same. The interface name could be different in different implementations instead. 
I mean I do not want to create confusion neither I want to block the interface name.

In my own implementation I named the interface as `IERC1363Spender` (not `ERC1363Spender`) so the returned value should be `IERC1363Spender(this).onApprovalReceived.selector` that is different from what you suggested also if it is the right value. I would leave the returned calculated value method to the implementer. He MUST return the `bytes4(keccak256("onApprovalReceived(address,uint256,bytes)"))` value.

Note that I used the one you suggested in [my implementation](https://github.com/vittominacori/erc1363-payable-token/blob/master/contracts/payment/ERC1363Payable.sol#L75) but I prefer to leave it as is in the spec.